File: app/src/main/java/com/mona/moviesapp/popular_people/PopularListScreen/model/ListModel.java
package com.mona.moviesapp.popular_people.PopularListScreen.model;

import com.mona.moviesapp.popular_people.PopularListScreen.controller.ListController;

public class ListModel {

    ListController listController;
    DataNetwork dataNetwork = new DataNetwork(this, listController);

    public ListModel(ListController listController) {
        this.listController = listController;
        this.dataNetwork = dataNetwork;
    }

    public void geturl(String popularurl){
        DataNetwork dataNetwork = new DataNetwork(this, listController);
        dataNetwork.execute(popularurl);
    }
}

File: app/src/main/java/com/mona/moviesapp/popular_people/PopularListScreen/model/ListModel.kt
package com.mona.moviesapp.popular_people.PopularListScreen.model

import com.mona.moviesapp.popular_people.PopularListScreen.controller.ListController

class ListModel(internal var listController: ListController) {
    internal var dataNetwork = DataNetwork(this, listController)

    init {
        this.dataNetwork = dataNetwork
    }

    fun geturl(popularurl: String) {
        val dataNetwork = DataNetwork(this, listController)
        dataNetwork.execute(popularurl)
    }
}
